internal/gwsidecar/gwsidecar.go

```diff
@@ -51,16 +51,27 @@ type GwConnectionContext struct {
 	RemoteSliceGwNsmSubnet string
 }
 
+type WorkerClientRpc interface {
+	GetStatus(ctx context.Context, serverAddr string) (*GwStatus, error)
+}
+
+type workerClientConfig struct {
+	Client sidecar.GwSidecarServiceClient
+}
+
+func NewWorkerGWSidecarClientProvider() (*workerClientConfig, error) {
+	return &workerClientConfig{}, nil
+}
+
 // GetStatus retrieves sidecar status
-func GetStatus(ctx context.Context, serverAddr string) (*GwStatus, error) {
+func (worker workerClientConfig) GetStatus(ctx context.Context, serverAddr string) (*GwStatus, error) {
 	conn, err := grpc.Dial(serverAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
 	if err != nil {
 		return nil, err
 	}
 	defer conn.Close()
-
-	client := sidecar.NewGwSidecarServiceClient(conn)
-	res, err := client.GetStatus(ctx, &empty.Empty{})
+	worker.Client = sidecar.NewGwSidecarServiceClient(conn)
+	res, err := worker.Client.GetStatus(ctx, &empty.Empty{})
```

Comment:
why are we storing the `Client` in `worker` struct, if it not used outside this scope? 
Should we run `sidecar.NewGwSidecarServiceClient(conn)` at the time of creating the struct and reuse the client everywhere?

Reply:
@YachikaRalhan

Reply:
The problem is client type is different for the all components like a gateway, router, etc. So that is why each component has it's own interface and own client type.

Reply:
My question is why are we assigning grpc client to `worker.Client` variable?
This variable is not used anywhere outside this function scope.